doc/intro sphinx (#46)
pyvista/pyvista-tutorial: tutorial/01_basic/README.rst

Basic usage
===========

Reading and plotting 3D data using ``examples`` module and external files.  (10
min for talk, 10 min for exercise)

.. tip::

    This section of the tutorial was adopted from the `Basic API Usage
    <https://docs.pyvista.org/user-guide/simple.html>`_ chapter of the PyVista
    documentation.

> Using Exising Data
> ~~~~~~~~~~~~~~~~~~
> There are two main ways of getting data into PyVista: creating it yourself from
> scratch or loading the dataset from any one of the `compatible file formats
> <https://docs.pyvista.org/api/readers/index.html>`_. Since we're starting out
> here, let's just load a file.

Reviewer: ```suggestion
Using Existing Data
~~~~~~~~~~~~~~~~~~~
There are two main ways of getting data into PyVista: creating it yourself from
scratch or loading the dataset from any one of the `compatible file formats
<https://docs.pyvista.org/api/readers/index.html>`_. Since we're starting out
here, let's just load a file.
```

If you have a dataset handy, like a surface model, point cloud, or VTK file,
you can use that. If you don't have something immediately available, PyVista
has a variety of files you can download in its `pyvista.examples
<https://docs.pyvista.org/api/examples/_autosummary/pyvista.examples.downloads.html>`_
module.

Here's a very basic dataset you can download.

.. jupyter-execute::
   :hide-code:

   # Configure for panel
   import pyvista
   pyvista.set_jupyter_backend('panel')
   pyvista.global_theme.background = 'white'
   pyvista.global_theme.axes.show = False
   pyvista.global_theme.smooth_shading = True
   pyvista.global_theme.antialiasing = True

.. jupyter-execute::

   from pyvista import examples
   dataset = examples.download_saddle_surface()
   dataset

Note how this is a :class:`pyvista.PolyData`, which is effectively a surface
dataset containing points, lines, and/or faces. We can immediately plot this with:

.. jupyter-execute::

   dataset.plot(color='tan')

> Note how this is a fairly basic plot. You can change how its plotted by adding

Author: ```suggestion
This is a fairly basic plot. You can change how its plotted by adding
```

parameters as ``show_edges=True`` or changing the color by setting ``color`` to
a different value. All of this is described in PyVista's API documentation in
:func:`pyvista.plot`, but for now let's take a look at another dataset. This
one is a volumetric dataset.

.. jupyter-execute::

   dataset = examples.download_frog()
   dataset

This is a :class:`pyvista.UniformGrid`, which is a dataset containing a uniform
set of points with consistent spacing. When we plot this dataset, we have the
option of enabling volumetric plotting, which plots individual cells based on
the content of the data associated with those cells.

.. jupyter-execute::

   dataset.plot(volume=True)

.. note::
   One thing you might have noticed by now is that the plots here in the online

>    tutorial may look slightly different depending on how you're plotting them

Author: ```suggestion
   tutorial may look slightly different than your plots depending on how you're plotting them
```

>    on your computer. This will be depending on your ``jupyter_backend``, or if

Author: ```suggestion
   on your computer. This depends on your ``jupyter_backend``, or if
```

>    One thing you might have noticed by now is that the plots here in the online
>    tutorial may look slightly different depending on how you're plotting them
>    on your computer. This will depending on your ``jupyter_backend``, or if
>    you're even using a jupyter notebook. As you're playing around with these
>    examples, feel free to change settings like disabling (or enabling)
>    ``notebook``, or using a different plotting backend for displaying within
>    your notebook (if applicable).

Reviewer: ```suggestion
   One thing you might have noticed by now is that the plots here in the online
   tutorial may look slightly different depending on how you're plotting them
   on your computer. This will be depending on your ``jupyter_backend``, or if
   you're even using a jupyter notebook. As you're playing around with these
   examples, feel free to change settings like disabling (or enabling)
   ``notebook``, or using a different plotting backend for displaying within
   your notebook (if applicable).
```

Read from a file
~~~~~~~~~~~~~~~~
You can read datasets directly from a file if you have access to it locally on
your computer. This can be one of the many file formats that VTK supports, and

> many more that it doesn't thanks to libraries like `meshio

Author: ```suggestion
many more that it doesn't as PyVista can rely on libraries like `meshio
```

> You can read datasets directly from a file if you have access to it locally on
> your computer. This can be one of the many file formats that VTK supports, and
> many more that it doesn't thanks to libraries like `meshio
> <https://github.com/nschloe/meshio>`_.

Reviewer: ```suggestion
You can read datasets directly from a file if you have access to it locally on
your computer. This can be one of the many file formats that VTK supports, and
many more that it doesn't thank libraries like `meshio
<https://github.com/nschloe/meshio>`_.
```

Author: This is correct.

In the following example, we load VTK's iron protein dataset `ironProt.vtk
<https://github.com/naucoin/VTKData/blob/master/Data/ironProt.vtk>`_ from a
file using :func:`pyvista.read`.

.. jupyter-execute::

   import pyvista as pv
   dataset = pv.read('ironProt.vtk')
   dataset

Notice that this is again a :class:`pyvista.UniformGrid` and we can plot it
volumetrically with:

.. jupyter-execute::

   dataset.plot(volume=True)

Exercises
---------

> Try out downloading and plotting datasets on your own with the following example. 

Author: ```suggestion
Try out downloading and plotting datasets on your own with the following example.
```
